Controllers/OrdersController.cs

```diff
@@ -0,0 +1,109 @@
+        }
+
+        [HttpGet("ordersf")] //return a full list of orders and products
+        public async Task<ActionResult<IEnumerable<GetOrdersDTO<ProductBalanceInterDTO>>>> GetOrders2()
+        {
+            if (!IsValidApiRequest())
+            {
+                return ApiBadRequest("Invalid Headers!");
+            }
+
+            var orders = await Context.Order
+                .Select(o => new OrdersDTO<ProductBalanceInterDTO>(){
+                    Id = o.Id,
+                    OrderTime = o.OrderTime,
+                    DeliveryTime = o.DeliveryTime,
+                    AddressFrom = o.AddressFrom,
+                    AddressTo = o.AddressTo,
+                    Status = o.Status,
+                    Products = Context.ProductBalance
+                                .Where(x => x.Order.Id == o.Id)
+                                .Select(x => new ProductBalanceInterDTO() { 
+                                        Id = x.Id,
+                                        ExpirationDate = x.ExpirationDate,
+                                        Price = x.Price,
+                                        Medicament = Context.Medicaments.Where(y => y.Id == x.Id).First().Name,
+                                        Transaction = Context.Transaction.Where(y => y.Id == x.Id)
+                                                            .Select(y => new TransactionInterDTO() { 
+                                                                Id = y.Id,
+                                                                Sum = y.Sum,
+                                                                Date = y.Date,
+                                                                Method = y.Method,
+                                                                Employee = Context.Employees.Where(z => z.PersonalCode == y.Pharmacist.PersonalCode).First().FirstName
+                                                            }).First(),
+                                        Pharmacy = Context.Pharmacy.Where(y => y.Id == x.Id).First().Address,
+                                        Warehouse = Context.Warehouse.Where(y => y.Id == x.Id).First().Address,
+                                        Provider = Context.MedicineProvider.Where(y => y.Id == x.Id).First().Name
+
+                                }).ToList()
+                }).ToListAsync();           
+
+            return Ok(new GetOrdersDTO<ProductBalanceInterDTO>(orders));
+        }
+
+        [HttpPost("registerprovider")]
+        public async Task<IActionResult> RegisterProvider(MedicineProvider model)
+        {
+            if (!IsValidApiRequest())
+            {
+                return ApiBadRequest("Invalid Headers!");
+            }
+
+            if (!Context.MedicineProvider.Select(x => x.Name).Contains(model.Name))
+            {
+
+            }
+
+            var message = new MessageDTO("Hello World!");
+            return Ok(message);
+        }
+    }
```

Comment:
This method doesn't have any logic. Is it really required?

Reply:
Yes, this method isn't needed.

Reply:
Indeed